Test failure in armel: parser panic: runtime error: integer divide by zero

Hi,

I am getting build failures in armel due to a test [failing](https://buildd.debian.org/status/fetch.php?pkg=prometheus&arch=armel&ver=1.1.3%2Bds-1&stamp=1475434773) with an "unexpected error", that turns out to be a division by zero in the parser:

```
time="2016-10-02T18:56:22Z" level=error msg="parser panic: runtime error: integer divide by zero
goroutine 386 [running]:
github.com/prometheus/prometheus/promql.(*evaluator).recover(0x135a2b30, 0x1084bb70)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:557 +0xb8
panic(0x3d13d0, 0x10828010)
    /usr/lib/go-1.7/src/runtime/panic.go:458 +0x454
github.com/prometheus/prometheus/promql.vectorElemBinop(0x17, 0x0, 0x40a45000, 0x0, 0x45000000, 0x13d93638, 0x0, 0x2)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:1060 +0x1f0
github.com/prometheus/prometheus/promql.(*evaluator).vectorScalarBinop(0x135a2b30, 0x17, 0x13d93630, 0x2, 0x2, 0x135a2c60, 0x135a0000, 0x0, 0x0, 0x0)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:991 +0x104
github.com/prometheus/prometheus/promql.(*evaluator).eval(0x135a2b30, 0x579200, 0x13f290c0, 0x0, 0x0)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:669 +0xe2c
github.com/prometheus/prometheus/promql.(*evaluator).Eval(0x135a2b30, 0x579200, 0x13f290c0, 0x0, 0x0, 0x0, 0x0)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:630 +0x94
github.com/prometheus/prometheus/promql.(*Engine).execEvalStmt(0x135a6360, 0xb5380000, 0x1474ca00, 0x1084be88, 0x13642980, 0x0, 0x0, 0x0, 0x0)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:383 +0x250
github.com/prometheus/prometheus/promql.(*Engine).exec(0x135a6360, 0x1084be88, 0x0, 0x0, 0x0, 0x0)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:357 +0x4b4
github.com/prometheus/prometheus/promql.(*query).Exec(0x1084be88, 0x0)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:196 +0x28
github.com/prometheus/prometheus/promql.(*Test).exec(0x1231a4b0, 0x576c90, 0x1232f920, 0x0, 0x0)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/test.go:466 +0x29c
github.com/prometheus/prometheus/promql.(*Test).Run(0x1231a4b0, 0x0, 0x0)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/test.go:444 +0xa8
github.com/prometheus/prometheus/promql.TestEvaluations(0x1080aa80)
    /«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/promql_test.go:31 +0x274
testing.tRunner(0x1080aa80, 0x45ad90)
    /usr/lib/go-1.7/src/testing/testing.go:610 +0xa4
created by testing.(*T).Run
    /usr/lib/go-1.7/src/testing/testing.go:646 +0x304
" source="engine.go:559" 
```

Full test log output:

```
=== RUN   TestEvaluations
time="2016-10-02T18:56:19Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:19Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:19Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:19Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:19Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:19Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:19Z" level=info msg="Done checkpointing in-memory metrics and chunks in 822.831µs." source="persistence.go:572" 
time="2016-10-02T18:56:19Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:19Z" level=info msg="Done checkpointing fingerprint mappings in 134.238µs." source="persistence.go:1394" 
time="2016-10-02T18:56:19Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:19Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:19Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:19Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:19Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:19Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:19Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:19Z" level=info msg="Done checkpointing in-memory metrics and chunks in 372.957µs." source="persistence.go:572" 
time="2016-10-02T18:56:19Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:19Z" level=info msg="Done checkpointing fingerprint mappings in 137.358µs." source="persistence.go:1394" 
time="2016-10-02T18:56:19Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:19Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:19Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:19Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:19Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:19Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:19Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:19Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:19Z" level=info msg="Done checkpointing in-memory metrics and chunks in 644.914µs." source="persistence.go:572" 
time="2016-10-02T18:56:19Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:19Z" level=info msg="Done checkpointing fingerprint mappings in 139.039µs." source="persistence.go:1394" 
time="2016-10-02T18:56:19Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 867.511µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 208.958µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 856.712µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 267.478µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 867.631µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 493.555µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 365.716µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 163.398µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 318.316µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 144.079µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 335.717µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 145.159µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 312.157µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 142.558µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 301.797µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 192.878µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 637.834µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 205.358µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 313.077µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 164.959µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 333.957µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 143.639µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 343.196µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 223.158µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 328.637µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 155.558µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:20Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:20Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:20Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:20Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing in-memory metrics and chunks in 692.953µs." source="persistence.go:572" 
time="2016-10-02T18:56:20Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:20Z" level=info msg="Done checkpointing fingerprint mappings in 141.318µs." source="persistence.go:1394" 
time="2016-10-02T18:56:20Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:20Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:20Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 530.635µs." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 147.079µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:21Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:21Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 460.675µs." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 159.359µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:21Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:21Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 410.796µs." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 261.518µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:21Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:21Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 358.676µs." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 145.279µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:21Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:21Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 214.357µs." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 142.319µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:21Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:21Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 1.669344ms." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 146.318µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:21Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:21Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 1.618744ms." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 141.438µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:21Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:21Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 389.716µs." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 140.679µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:21Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:21Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 552.954µs." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 228.158µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:21Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:21Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 279.397µs." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 142.399µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:21Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:21Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:21Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:21Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:21Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:21Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing in-memory metrics and chunks in 249.197µs." source="persistence.go:572" 
time="2016-10-02T18:56:21Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:21Z" level=info msg="Done checkpointing fingerprint mappings in 145.839µs." source="persistence.go:1394" 
time="2016-10-02T18:56:21Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:22Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:22Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:22Z" level=error msg="parser panic: runtime error: integer divide by zero\ngoroutine 386 [running]:\ngithub.com/prometheus/prometheus/promql.(*evaluator).recover(0x135a2b30, 0x1084bb70)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:557 +0xb8\npanic(0x3d13d0, 0x10828010)\n\t/usr/lib/go-1.7/src/runtime/panic.go:458 +0x454\ngithub.com/prometheus/prometheus/promql.vectorElemBinop(0x17, 0x0, 0x40a45000, 0x0, 0x45000000, 0x13d93638, 0x0, 0x2)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:1060 +0x1f0\ngithub.com/prometheus/prometheus/promql.(*evaluator).vectorScalarBinop(0x135a2b30, 0x17, 0x13d93630, 0x2, 0x2, 0x135a2c60, 0x135a0000, 0x0, 0x0, 0x0)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:991 +0x104\ngithub.com/prometheus/prometheus/promql.(*evaluator).eval(0x135a2b30, 0x579200, 0x13f290c0, 0x0, 0x0)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:669 +0xe2c\ngithub.com/prometheus/prometheus/promql.(*evaluator).Eval(0x135a2b30, 0x579200, 0x13f290c0, 0x0, 0x0, 0x0, 0x0)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:630 +0x94\ngithub.com/prometheus/prometheus/promql.(*Engine).execEvalStmt(0x135a6360, 0xb5380000, 0x1474ca00, 0x1084be88, 0x13642980, 0x0, 0x0, 0x0, 0x0)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:383 +0x250\ngithub.com/prometheus/prometheus/promql.(*Engine).exec(0x135a6360, 0x1084be88, 0x0, 0x0, 0x0, 0x0)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:357 +0x4b4\ngithub.com/prometheus/prometheus/promql.(*query).Exec(0x1084be88, 0x0)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/engine.go:196 +0x28\ngithub.com/prometheus/prometheus/promql.(*Test).exec(0x1231a4b0, 0x576c90, 0x1232f920, 0x0, 0x0)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/test.go:466 +0x29c\ngithub.com/prometheus/prometheus/promql.(*Test).Run(0x1231a4b0, 0x0, 0x0)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/test.go:444 +0xa8\ngithub.com/prometheus/prometheus/promql.TestEvaluations(0x1080aa80)\n\t/«BUILDDIR»/prometheus-1.1.3+ds/build/src/github.com/prometheus/prometheus/promql/promql_test.go:31 +0x274\ntesting.tRunner(0x1080aa80, 0x45ad90)\n\t/usr/lib/go-1.7/src/testing/testing.go:610 +0xa4\ncreated by testing.(*T).Run\n\t/usr/lib/go-1.7/src/testing/testing.go:646 +0x304\n" source="engine.go:559" 
time="2016-10-02T18:56:22Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:22Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:22Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:22Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:22Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:22Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:22Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:22Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:22Z" level=info msg="Done checkpointing in-memory metrics and chunks in 868.672µs." source="persistence.go:572" 
time="2016-10-02T18:56:22Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:22Z" level=info msg="Done checkpointing fingerprint mappings in 142.558µs." source="persistence.go:1394" 
time="2016-10-02T18:56:22Z" level=info msg="Local storage stopped." source="storage.go:406" 
time="2016-10-02T18:56:22Z" level=info msg="Loading series map and head chunks..." source="storage.go:358" 
time="2016-10-02T18:56:22Z" level=info msg="0 series loaded." source="storage.go:363" 
time="2016-10-02T18:56:22Z" level=info msg="Stopping local storage..." source="storage.go:381" 
time="2016-10-02T18:56:22Z" level=info msg="Stopping maintenance loop..." source="storage.go:383" 
time="2016-10-02T18:56:22Z" level=info msg="Maintenance loop stopped." source="storage.go:1143" 
time="2016-10-02T18:56:22Z" level=info msg="Stopping series quarantining..." source="storage.go:387" 
time="2016-10-02T18:56:22Z" level=info msg="Series quarantining stopped." source="storage.go:1554" 
time="2016-10-02T18:56:22Z" level=info msg="Stopping chunk eviction..." source="storage.go:391" 
time="2016-10-02T18:56:22Z" level=info msg="Chunk eviction stopped." source="storage.go:963" 
time="2016-10-02T18:56:22Z" level=info msg="Checkpointing in-memory metrics and chunks..." source="persistence.go:548" 
time="2016-10-02T18:56:22Z" level=info msg="Done checkpointing in-memory metrics and chunks in 511.114µs." source="persistence.go:572" 
time="2016-10-02T18:56:22Z" level=info msg="Checkpointing fingerprint mappings..." source="persistence.go:1371" 
time="2016-10-02T18:56:22Z" level=info msg="Done checkpointing fingerprint mappings in 158.759µs." source="persistence.go:1394" 
time="2016-10-02T18:56:22Z" level=info msg="Local storage stopped." source="storage.go:406" 
--- FAIL: TestEvaluations (2.76s)
    promql_test.go:33: error running test testdata/operators.test: error evaluating query: unexpected error
FAIL
exit status 1
FAIL    github.com/prometheus/prometheus/promql 2.938s
```

--- a/promql/engine.go
+++ b/promql/engine.go
@@ -1012,7 +1012,7 @@ func scalarBinop(op itemType, lhs, rhs model.SampleValue) model.SampleValue {
 	case itemPOW:
 		return model.SampleValue(math.Pow(float64(lhs), float64(rhs)))
 	case itemMOD:
-		if rhs != 0 {
+		if int(rhs) != 0 {
 			return model.SampleValue(int(lhs) % int(rhs))
 		}
 		return model.SampleValue(math.NaN())
@@ -1046,7 +1046,7 @@ func vectorElemBinop(op itemType, lhs, rhs model.SampleValue) (model.SampleValue
 	case itemPOW:
 		return model.SampleValue(math.Pow(float64(lhs), float64(rhs))), true
 	case itemMOD:
-		if rhs != 0 {
+		if int(rhs) != 0 {
 			return model.SampleValue(int(lhs) % int(rhs)), true
 		}
 		return model.SampleValue(math.NaN()), true

--- a/promql/testdata/operators.test
+++ b/promql/testdata/operators.test
@@ -34,6 +34,10 @@ eval instant at 50m SUM(http_requests) BY (job) % 3
 	{job="api-server"} 1
 	{job="app-server"} 2
 
+eval instant at 50m SUM(http_requests) BY (job) % 0.3
+	{job="api-server"} NaN
+	{job="app-server"} NaN
+
 eval instant at 50m SUM(http_requests) BY (job) ^ 2
 	{job="api-server"} 1000000
 	{job="app-server"} 6760000
@@ -347,3 +351,13 @@ eval instant at 5m metricA + ignoring() metricB
 
 eval instant at 5m metricA + metricB
   {baz="meh"} 7
+
+clear
+
+load 5m
+  finite{foo="bar"} 42
+  almost_zero{foo="bar"} 0.123
+
+# MOD by "almost zero" with vector.
+eval instant at 5m finite % almost_zero
+  {foo="bar"} NaN

